### PR TITLE
add extension plugin framework for nodeslo

### PR DESCRIPTION
**Pull request by zwzhang0107**
Signed-off-by: 佑祎 <zzw261520@alibaba-inc.com>

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/util/system/util_test_tool.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ func NewFileTestUtil(t *testing.T) *FileTestUtil {
 	return &FileTestUtil{TempDir: tempDir, t: t}
 }
 
+func (c *FileTestUtil) Cleanup() {
+	os.RemoveAll(c.TempDir)
```
Comment (saintube) [inline pkg/util/system/util_test_tool.go:52]: Do we still need the `Cleanup()`? Directories created by `t.TempDir()` are cleaned automatically when the unit test finished.

Comment (zwzhang0107) [reply pkg/util/system/util_test_tool.go:52]: keep this function only for ut

**Review by jasonliu747 — COMMENTED** [review]
The diff coverage is a little bit low. Would you mind adding more UTs for this PR?

**Pull request merged**